src/commands/moderation/prefixCommands/functions/addCommand.ts

```diff
@@ -0,0 +1,178 @@
+            value: embedColor || '',
+        },
+    ],
+    footer: { text: `Command ID: ${commandId}` },
+    color: Colors.Green,
+});
+
+const noModLogs = makeEmbed({
+    title: 'Prefix Commands - Add Command - No Mod Log',
+    description: 'I can\'t find the mod logs channel. Please check the channel still exists.',
+    color: Colors.Red,
+});
+
+export async function handleAddPrefixCommand(interaction: ChatInputCommandInteraction<'cached'>) {
+    await interaction.deferReply({ ephemeral: true });
+
+    const conn = getConn();
+    if (!conn) {
+        await interaction.followUp({ embeds: [noConnEmbed], ephemeral: true });
+        return;
+    }
+
+    const name = interaction.options.getString('name')?.toLowerCase().trim()!;
+    const category = interaction.options.getString('category')!;
+    const description = interaction.options.getString('description')!;
+    const aliasesString = interaction.options.getString('aliases')?.toLowerCase().trim() || '';
+    const aliases = aliasesString !== '' ? aliasesString.split(',') : [];
+    const isEmbed = interaction.options.getBoolean('is_embed') || false;
+    const embedColor = interaction.options.getString('embed_color') || '';
+    const moderator = interaction.user;
+
+    const nameRegex = /^[\w-]+$/;
+    if (!nameRegex.test(name)) {
+        await interaction.followUp({ embeds: [wrongFormatEmbed(name)], ephemeral: true });
+        return;
+    }
+    for (const alias of aliases) {
+        if (!nameRegex.test(alias)) {
+            // eslint-disable-next-line no-await-in-loop
+            await interaction.followUp({ embeds: [wrongFormatEmbed(alias)], ephemeral: true });
+            return;
+        }
+    }
+
+    // Check if command name and alias are unique, additionally check if they do not exist as a version alias.
+    const foundCommandName = await PrefixCommand.findOne({
+        $or: [
+            { name },
+            { name: { $in: aliases } },
+            { aliases: name },
+            { aliases: { $in: aliases } },
+        ],
+    });
+    if (foundCommandName) {
+        await interaction.followUp({ embeds: [alreadyExistsEmbed(name, `${name} already exists as a command or alias, or one of the aliases already exists as a command or alias.`)], ephemeral: true });
+        return;
+    }
+    const foundVersion = await PrefixCommandVersion.findOne({
+        $or: [
+            { alias: name },
+            { alias: { $in: aliases } },
+        ],
+    });
+    if (foundVersion || name.toLowerCase() === 'generic' || aliases.includes('generic')) {
+        await interaction.followUp({ embeds: [alreadyExistsEmbed(name, `${name} already exists as a version alias, or one of the aliases already exists as a version alias.`)], ephemeral: true });
+        return;
+    }
+
+    //Check if the mod logs channel exists
+    const modLogsChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS) as TextChannel;
+    if (!modLogsChannel) {
```

Comment:
```suggestion
    let modLogsChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS);
    if (!modLogsChannel || !modLogsChannel.isTextBased()) {
        modLogsChannel = null;
```
Might be safer to actually check if the channel is a text channel? Especially if we want to offer the codebase to other projects. Then, if it's not a text channel set it to null to prevent accidental attempts to send into a non-text channel.